src/pretix/base/models/event.py

```diff
@@ -917,11 +918,16 @@ def _walk_rules(rules):
             s.object = self
             s.pk = None
             if s.value.startswith('file://'):
-                fi = default_storage.open(s.value[7:], 'rb')
+                fi = default_storage.open(s.value[len('file://'):], 'rb')
                 nonce = get_random_string(length=8)
+                if s.key == 'mail_attachment_new_order':
+                    fname_base = clean_filename(os.path.basename(s.value[len('file://'):]))
```

Comment:
AFAIK and in my tests I did not need to remove `file://` when using `os.path.basename`. So maybe shorten to this?

```suggestion
                    fname_base = clean_filename(os.path.basename(s.value))
```